app/controllers/application_controller.rb

```diff
@@ -91,8 +91,13 @@ def save_and_audit(model, attributes)
     def check_authorization
       if !@incident.authorized_to_view?(@current_user)
         raise BridgeExceptions::UnathorizedToView.new
-      elsif !@incident.authorized_to_edit?(@current_user) && params[:action] != 'review'
-        redirect_to review_incident_path(@incident)
+      elsif !@incident.authorized_to_edit?(@current_user)
+        # Redirect to review page if possible.
```

Comment:
Hmm, I can't actually think of any reasonable scenario where the viewable+uneditable+incomplete case would happen, at least given the way that `authorized_to_edit?` and `authorized_to_view?` currently work.

My change here is more of a just-in-case measure, to prevent a infinite redirect loop from arising in a potential pathological situation.